### PR TITLE
Fix regression in Get-PSCallStack

### LINKED ISSUE #11208: Get-PSCallStack's Arguments script property contains no useful information
The `.Arguments` property on the `System.Management.Automation.CallStackFrame` instances output by `Get-PSCallStack` is a script property that - I presume - is meant to reflect the arguments passed.

The property isn't documented, but it seems to contain no useful information, seemingly containing `$null` in any non-global scope.

While you con obtain argument information via the type-native `.InvocationInfo` property, I assume there is a purpose to `.Arguments` that it currently doesn't fulfill.

# Steps to reproduce

```powershell
& { (Get-PSCallStack)[0].Arguments } 'foo'  | Should -match 'foo'
```

# Expected behavior

The test should pass.

# Actual behavior

The test fails, because `.Arguments` is `$null`:

```none
Expected regular expression 'foo' to match $null, but it did not match.
```

# Environment data

```none
PowerShell Core 7.0.0-preview.6
```

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Fix #11208

## PR Context

The regression comes from #6950

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs [hunk 1]
```diff
@@ -6058,7 +6058,7 @@ private void Process_Types_Ps1Xml(string filePath, ConcurrentBag<string> errors)
           {
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Get-PSCallStack DRT Unit Tests.Verify that the script block of a trap statement shows up on the call stack
```powershell
Expected 77, but got 80.
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Get-PSCallStack.Tests.ps1: line 85
85:         $results[0].InvocationInfo.ScriptLineNumber | Should -Be 77
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Get-PSCallStack DRT Unit Tests.Verify that the script block of a trap statement shows up on the call stack
```powershell
Expected 77, but got 80.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Get-PSCallStack.Tests.ps1: line 85
85:         $results[0].InvocationInfo.ScriptLineNumber | Should -Be 77
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Get-PSCallStack DRT Unit Tests.Verify that the script block of a trap statement shows up on the call stack
```powershell
Expected 77, but got 80.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-PSCallStack.Tests.ps1: line 85
85:         $results[0].InvocationInfo.ScriptLineNumber | Should -Be 77
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Get-PSCallStack DRT Unit Tests.Verify that the script block of a trap statement shows up on the call stack
```powershell
Expected 77, but got 80.
at <ScriptBlock>, /Users/runner/runners/2.160.1/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Get-PSCallStack.Tests.ps1: line 90
90:         $results[1].InvocationInfo.ScriptLineNumber | Should -Be 77
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Get-PSCallStack DRT Unit Tests.Verify that the script block of a trap statement shows up on the call stack
```powershell
Expected 77, but got 80.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Get-PSCallStack.Tests.ps1: line 90
90:         $results[1].InvocationInfo.ScriptLineNumber | Should -Be 77
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Get-PSCallStack DRT Unit Tests.Verify that the script block of a trap statement shows up on the call stack
```powershell
Expected 77, but got 80.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Get-PSCallStack.Tests.ps1: line 90
90:         $results[1].InvocationInfo.ScriptLineNumber | Should -Be 77
```

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Get-PSCallStack.Tests.ps1 [hunk 2]
```diff
@@ -79,11 +82,15 @@ Describe "Get-PSCallStack DRT Unit Tests" -Tags "CI" {
...
 
         $results[1].Command | Should -Be $scriptFileName
         $results[1].ScriptName | Should -Be $scriptFilePath
         $results[1].ScriptLineNumber | Should -Be 7
-        $results[1].InvocationInfo.ScriptLineNumber | Should -Be 77
+        $results[1].InvocationInfo.ScriptLineNumber | Should -Be 80
+    }
+
+    It "Get-PSCallStack returns Arguments" {
+        & { (Get-PSCallStack)[0].Arguments } 'foo'  | Should -Match 'foo'
```
Comment (BrucePay) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Get-PSCallStack.Tests.ps1]: Please also add a test for the case where the scriptblock has formal parameters:
```PowerShell
 & { param ($x)  (Get-PSCallStack)[0].Arguments } 'foo' 
```
and another where multiple arguments are passed to the scriptblock:
```powershell
 & { (Get-PSCallStack)[0].Arguments } 'foo'  'bar'
```

Comment (iSazonov) [reply test/powershell/Modules/Microsoft.PowerShell.Utility/Get-PSCallStack.Tests.ps1]: Done.

File: src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/TypeTable_Types_Ps1Xml.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Test-Connection.Error occurred in Describe block
```powershell
Exception calling "GetHostEntry" with "1" argument(s): "Name or service not known"
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:15 char:9
```

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**